src/lib.rs

```diff
@@ -114,7 +100,12 @@ pub fn decode_str(input: &str, mode: ParseMode) -> Result<Vec<u8>, QuotedPrintab
 /// to IETF RFC 2045, section 6.7 for details on what constitutes valid and
 /// invalid input, and what a "robust" implementation would do in the face of
 /// invalid input.
-pub fn decode(input: &[u8], mode: ParseMode) -> Result<Vec<u8>, QuotedPrintableError> {
+#[inline(always)]
+pub fn decode<R: AsRef<[u8]>>(input: R, mode: ParseMode) -> Result<Vec<u8>, QuotedPrintableError> {
```

Comment:
Please update the README.md file in this change to reflect the change in signature, and the removal of the decode_str function.